BUG: Lineage image fails to start with class not found exception

## Description
The docker image fails to start due to a ClassNotFound exception, likely introduced by #699. Note that Kafka Clients was added at the `test` scope instead of the default compile scope.

## Steps to Reproduce
Clear, specific, and detailed steps taken to enable reproduction of the bug for investigation.
1. Start the `aissemble-data-lineage-http-consumer` image
    ```sh
    docker run --rm ghcr.io/boozallen/aissemble-data-lineage-http-consumer:1.13.0-SNAPSHOT
    ```
2. Note that the image crashes

### Expected Behavior
The Quarkus service is started and ready to listen for messages.

### Actual Behavior
The Docker image crashes.

## Definition of Done
 - [x] add an IT to the image to ensure that it starts properly
 - [x] fix the cause of the crash

## Test Steps
 - Verify the [latest build](https://github.com/boozallen/aissemble/actions/runs/17982005688) succeeded

## Additional Context
Stack Trace:
```log
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:70)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	... 15 more
Caused by: java.lang.ClassNotFoundException: org.apache.kafka.clients.consumer.Consumer
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:114)
	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:72)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at io.smallrye.reactive.messaging.kafka.KafkaCDIEvents_Bean.<init>(Unknown Source)
	at io.quarkus.arc.setup.Default_ComponentsProvider.addBeans1(Unknown Source)
	at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Unknown Source)
	at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:128)
	at io.quarkus.arc.Arc.initialize(Arc.java:38)
	at io.quarkus.arc.runtime.ArcRecorder.initContainer(ArcRecorder.java:49)
	at io.quarkus.deployment.steps.ArcProcessor$initializeContainer1770303700.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.ArcProcessor$initializeContainer1770303700.deploy(Unknown Source)
```

--- a/extensions/extensions-data-lineage/extensions-data-lineage-http-consumer-service/pom.xml
+++ b/extensions/extensions-data-lineage/extensions-data-lineage-http-consumer-service/pom.xml
@@ -36,6 +36,10 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-health</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-smallrye-reactive-messaging</artifactId>
         </dependency>
         <dependency>
@@ -63,6 +67,10 @@
                     <artifactId>kafka-clients</artifactId>
                 </exclusion>
             </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.kafka</groupId>
+            <artifactId>kafka-clients</artifactId>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
@@ -114,11 +122,6 @@
             <groupId>io.openlineage</groupId>
             <artifactId>openlineage-java</artifactId>
             <version>${version.open.lineage.java}</version>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.kafka</groupId>
-            <artifactId>kafka-clients</artifactId>
-            <scope>test</scope>
         </dependency>
     </dependencies>
 

--- a/extensions/extensions-docker/aissemble-data-lineage-http-consumer/pom.xml
+++ b/extensions/extensions-docker/aissemble-data-lineage-http-consumer/pom.xml
@@ -80,4 +80,55 @@
         </plugins>
     </build>
 
+    <profiles>
+        <profile>
+            <id>integration-test</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>${group.fabric8.plugin}</groupId>
+                        <artifactId>docker-maven-plugin</artifactId>
+                        <configuration>
+                            <buildArchiveOnly>false</buildArchiveOnly>
+                        </configuration>
+                        <executions>
+                            <execution>
+                                <id>test-image</id>
+                                <phase>integration-test</phase>
+                                <configuration>
+                                    <images>
+                                        <image>
+                                            <run>
+                                                <wait>
+                                                    <http>
+                                                        <url>http://localhost:8080/q/health</url>
+                                                        <method>GET</method>
+                                                        <status>200</status>
+                                                    </http>
+                                                </wait>
+                                                <ports>
+                                                    <port>8080:8080</port>
+                                                </ports>
+                                            </run>
+                                        </image>
+                                    </images>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>exec-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>test-image-cleanup</id>
+                                <phase>post-integration-test</phase>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+
 </project>

